web/src/user_profile.js

```diff
@@ -395,7 +396,10 @@ export function hide_user_profile() {
 function on_user_profile_hide() {
     user_streams_list_widget = undefined;
     user_profile_subscribe_widget = undefined;
-    browser_history.exit_overlay();
+    const base = get_current_hash_category();
+    if (base === "user") {
+        browser_history.exit_overlay();
+    }
```

Comment:
Probably we need a comment here as well as to under what situation that might not be true.